src/pages/transactions/addTransaction.js

```diff
@@ -0,0 +1,102 @@
+import { useRouter } from "next/router";
+
+export default function TransactionForm({ onAddTransaction, onCancel }) {
+  const router = useRouter();
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+    onAddTransaction(data);
+    router.push("/");
+  }
+
+  return (
+    <div>
+      <h1>Expense</h1>
+      <form onSubmit={handleSubmit}>
+        <button type="button" onClick={onCancel}>Cancel</button>
+        <div>
```

Comment:
This button is currently not doing what he supposed to, because even though you receive an onCancel prop, you never pass it in the app.js. Actually there is no need the pass any kind of function for this button. An easy fix would be this: 
```js
<button type="button" onClick={( ) => {
router.push("/")
}>Cancel</button>

Reply:
Right! We've changed this accordingly.